OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -66,11 +66,19 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 
 				var actorInfo = self.World.Map.Rules.Actors[order.TargetString];
 				var queue = targetActor.TraitsImplementing<ProductionQueue>()
-					.FirstOrDefault(q => q.CanBuild(actorInfo) && q.CurrentItem() != null && q.CurrentItem().Item == order.TargetString && q.CurrentItem().RemainingTime == 0);
+					.FirstOrDefault(q => q.CanBuild(actorInfo) && q.AllQueued().Any(i => i.Item == order.TargetString && i.Done));
```

Comment:
Can't you return a keyvaluepair from here with item along? That new item lookup could end up redundant thenafter.